src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let preValue = initialValue;
+
+    if (arguments.length < 2) {
+      preValue = this[0];
+
+      for (let i = 1; i < this.length; i++) {
```

Comment:
Consider using a single for loop with a conditional to handle the case when 'initialValue' is undefined.